rules-reviewed/openshift/mail.windup.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0"?>
+<ruleset id="mail"
+    xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This is a ruleset for detecting Mail API usage when migrating an application to a cloud environment.
+        </description>
+        <dependencies>
+          <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final"/>
+          <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="java"/>
+        <sourceTechnology id="java-ee"/>
+        <targetTechnology id="cloud-readiness"/>
+        <tag>mail</tag>
+    </metadata>
+    <rules>
+        <rule id="mail-00000">
+            <when>
+                    <javaclass references="javax.mail.{*}">
+                        <location>CONSTRUCTOR_CALL</location>
+                        <location>IMPORT</location>
+                        <location>INHERITANCE</location>
+                        <location>VARIABLE_DECLARATION</location>
+                    </javaclass>
+            </when>
+            <perform>
+              <hint title="JavaMail: direct access of an SMTP server" effort="3" category-id="cloud-optional">
```

Comment:
If you want SMTP removed, then I would recommend that the rule be removed completely. The advice doesn't really generalize beyond sending mail via SMTP. Also, it really isn't clear to me why accessing an email server from a cloud environment should be verboten.

If the user does have to migrate away from direct email access to some other API, there will be much more effort than "1" and probably much more than "3". It doesn't tend to be that easy to just pull in another cloud service within the typical enterprise.